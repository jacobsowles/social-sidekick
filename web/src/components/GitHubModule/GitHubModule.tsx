import React, { FunctionComponent } from 'react';

import './GitHubModule.scss';

const GitHubModule: FunctionComponent = () => {
  return (
    <div className="service-module github-module">
      <h2>GitHub</h2>
    </div>
  );
};

export default GitHubModule;
