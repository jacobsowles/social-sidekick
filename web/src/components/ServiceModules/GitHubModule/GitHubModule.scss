.github-module {
  width: 100%;

  .button {
    border: 1px solid rgba(0, 0, 0, 0.1);
    color: #ffffff;
    padding: 6px 12px;
  }

  .button-default {
    background-color: #ffffff;
    border-color: rgba(0, 0, 0, 0.1);
    color: #333;

    &:hover {
      background-color: #e6e6e6;
      border-color: rgba(0, 0, 0, 0.1);
    }
  }

  .button-primary {
    background-color: #5cb85c;
    border-color: #4cae4c;

    &:hover {
      background-color: #449d44;
      border-color: #398439;
    }
  }

  input {
    width: calc(100% - 40px);
  }

  input,
  textarea {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 10px;
    padding: 10px;
  }

  svg {
    opacity: 0.8;
    width: 40px !important;
  }

  textarea {
    width: 100%;
  }

  .form-field {
    display: block;
  }

  .form-buttons {
    margin-top: 10px;

    button {
      margin-right: 10px;

      &:last-child {
        margin-right: 0;
      }
    }
  }
}
