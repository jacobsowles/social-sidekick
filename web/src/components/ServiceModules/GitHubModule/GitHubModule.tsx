import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React, { FunctionComponent } from 'react';

import './GitHubModule.scss';

const GitHubModule: FunctionComponent = () => {
  return (
    <form onSubmit={onSubmit} className="github-module">
      <div className="form-item">
        <textarea aria-label="Bio" maxLength={160} placeholder="Bio" />
      </div>

      <div className="form-item">
        <FontAwesomeIcon icon={['fas', 'users']} />
        <input type="text" aria-label="Company" placeholder="Company" />
      </div>

      <div className="form-item">
        <FontAwesomeIcon icon={['fas', 'map-marker-alt']} />
        <input type="text" aria-label="Location" placeholder="Location" />
      </div>

      <div className="form-item">
        <FontAwesomeIcon icon={['fas', 'link']} />
        <input type="text" aria-label="Website" placeholder="Website" />
      </div>

      <div className="form-buttons">
        <button className="button button-primary" type="submit">
          Submit
        </button>
        <button className="button button-default" type="reset">
          Cancel
        </button>
      </div>
    </form>
  );
};

const onSubmit = async (values: any) => {
  // TODO
};

const validate = (values: any) => {
  // TODO
};

export default GitHubModule;
