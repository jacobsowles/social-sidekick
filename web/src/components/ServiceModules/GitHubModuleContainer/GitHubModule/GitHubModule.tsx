import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React, { FunctionComponent } from 'react';

import Form from '@components/Form';
import LoadingSpinner from '@components/LoadingSpinner';
import TextArea from '@components/TextArea';
import TextInput from '@components/TextInput';
import './GitHubModule.scss';

export interface GitHubModuleProps {
  bio: string;
  blog: string;
  company: string;
  isLoaded: boolean;
  location: string;
  onCancel: (event: any) => void;
  onChange: (event: any, fieldName: string) => void;
  onSubmit: (event: any) => void;
}

const GitHubModule: FunctionComponent<GitHubModuleProps> = ({
  bio,
  blog,
  company,
  isLoaded,
  location,
  onCancel,
  onChange,
  onSubmit
}) => {
  return isLoaded ? (
    <Form.Wrapper className="github-module" onSubmit={onSubmit}>
      <Form.Field>
        <TextArea
          maxLength={160}
          onChange={event => onChange(event, 'bio')}
          placeholder={'Bio'}
          value={bio}
        />
      </Form.Field>

      <Form.Field>
        <FontAwesomeIcon icon={['fas', 'users']} />
        <TextInput
          onChange={event => onChange(event, 'company')}
          placeholder={'Company'}
          value={company}
        />
      </Form.Field>

      <Form.Field>
        <FontAwesomeIcon icon={['fas', 'map-marker-alt']} />
        <TextInput
          onChange={event => onChange(event, 'location')}
          placeholder={'Location'}
          value={location}
        />
      </Form.Field>

      <Form.Field>
        <FontAwesomeIcon icon={['fas', 'link']} />
        <TextInput
          onChange={event => onChange(event, 'blog')}
          placeholder={'Website'}
          value={blog}
        />
      </Form.Field>

      <div className="form-buttons">
        <button className="button button-primary" type="submit">
          Save
        </button>
        <button className="button button-default" onClick={onCancel} type="reset">
          Cancel
        </button>
      </div>
    </Form.Wrapper>
  ) : (
    <LoadingSpinner />
  );
};

export default GitHubModule;
