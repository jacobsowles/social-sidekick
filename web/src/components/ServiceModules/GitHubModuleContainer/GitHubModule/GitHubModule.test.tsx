import React from 'react';

import { init, shallow, toJson } from '@tests/component-test-base';
import GitHubModule, { GitHubModuleProps } from './GitHubModule';

describe('GitHubModule', () => {
  let props: GitHubModuleProps;
  init(() => <GitHubModule {...props} />, 'div');

  beforeEach(() => {
    props = {
      bio: undefined,
      blog: undefined,
      company: undefined,
      isLoaded: false,
      location: undefined,
      onCancel: jest.fn(),
      onChange: jest.fn(),
      onSubmit: jest.fn()
    };
  });

  it('should render the component without crashing', () => {
    toJson(shallow()).should.matchSnapshot();
  });
});
