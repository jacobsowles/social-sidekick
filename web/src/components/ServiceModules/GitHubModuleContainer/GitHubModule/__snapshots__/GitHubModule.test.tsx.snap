// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitHubModule should render the component without crashing 1`] = `
<LoadingSpinner
  size="4x"
/>
`;
